src/typescript/fable-splitter/index.ts

```diff
@@ -0,0 +1,172 @@
+
+    Object.keys(data.logs || {}).forEach((key) => {
+        const out = output(key);
+        ensureArray(data.logs[key]).forEach((log: string) => out(`${key} ==> ${log}`));
+    });
+
+    const code: string | undefined = undefined;
+    if (babelOptions && babelOptions.sourceMaps) {
+        babelOptions.sourceFileName = Path.relative(Process.cwd(), data.fileName.replace(/\\/g, "/"));
+    }
+    const result = Babel.transformFromAst(data, code, babelOptions);
+    return result;
+}
+
+async function transformAsync(path: string, options: FableCompilerOptions) {
+    // if already compiled, do nothing
+    let fullPath = Path.resolve(path).replace(/\\/g, "/");
+    if (alreadyCompiled.has(fullPath)) {
+        return undefined;
+    }
+    alreadyCompiled.add(fullPath);
+
+    let parsed: Babel.BabelFileResult | undefined;
+
+    // compile F#
+    if (FSHARP_EXT.test(path)) {
+        const fableMsg = JSON.stringify(Object.assign({}, options.fable, { path }));
+        const babelAst = await client.send(options.port, fableMsg);
+        parsed = compileAST(babelAst, options.babel);
+    } else {
+        // compile JS
+        fullPath = JAVASCRIPT_EXT.test(fullPath) ? fullPath : fullPath + ".js";
+        if (fs.existsSync(fullPath)) {
+            parsed = Babel.transformFileSync(fullPath, options.babel);
+        }
+    }
+
+    if (parsed) {
+        console.log(`Fable compiled: ${fullPath}`);
+
+        // if not a .fsproj, save as JS file
+        if (!FSPROJ_EXT.test(path)) {
+            // TODO: create common folder structure
+            const fileName = Path.basename(fullPath).replace(FSHARP_EXT, ".js");
+            const filePath = Path.resolve(Path.join(options.outDir, fileName));
+            const fileDir = Path.dirname(filePath);
+            if (!fs.existsSync(fileDir)) { fs.mkdirSync(fileDir); }
+            fs.writeFileSync(filePath, parsed.code);
+        }
+    }
+
+    return parsed;
+}
+
+async function transformLoopAsync(path: string, options: FableCompilerOptions) {
+    const result: any = await transformAsync(path, options);
+    if (result && result.ast && result.ast.program) {
+        const dir = Path.dirname(path);
+        const files = result.ast.program.body || [];
+        for (const file of files) {
+            if (file.source) {
+                const filePath = Path.join(dir, file.source.value);
+                await transformLoopAsync(filePath, options);
+            }
+        }
+    }
+}
+
+export default function fableCompiler(options: FableCompilerOptions) {
+    options = options || {};
+
+    const babelOptions = options.babel || {};
+    babelOptions.plugins = customPlugins.concat(babelOptions.plugins || []);
+
+    const fableOptions = options.fable || {};
+    fableOptions.path = fableOptions.path || "";
+    fableOptions.define = fableOptions.define || [];
+    fableOptions.plugins = fableOptions.plugins || [];
+    fableOptions.fableCore = fableOptions.fableCore || "../fable-core";
```

Comment:
fable-core JS files are now distributed in the Fable.Core nuget package so we can just use `null` as the default value.